File: stats_series_01_week_07.py
# Snippet 1: Introduction to Linear Regression
from sklearn.linear_model import LinearRegression

X, y = [[1], [2], [3], [4]], [3, 6, 9, 12] # Simple example
model = LinearRegression().fit(X, y)

# Snippet 2: Multiple Linear Regression
X_multi = [[1, 2], [2, 4], [3, 6], [4, 8]]
model_multi = LinearRegression().fit(X_multi, y)

# Snippet 3: Assumptions and Diagnostics
import statsmodels.api as sm
import matplotlib.pyplot as plt

X = sm.add_constant(X) # Adding intercept
model_stats = sm.OLS(y, X).fit()
residuals = model_stats.resid
plt.scatter(model_stats.predict(), residuals)
plt.hlines(0, min(model_stats.predict()), max(model_stats.predict()), colors='r')
plt.show()

# Snippet 4: Interpretation and Applications (Using the previous models)
slope = model.coef_
intercept = model.intercept_

File: stats_series_01_week_07.R
# Snippet 1: Introduction to Linear Regression
X <- c(1, 2, 3, 4)
y <- c(3, 6, 9, 12)
model <- lm(y ~ X)

# Snippet 2: Multiple Linear Regression
X_multi <- data.frame(X1 = c(1, 2, 3, 4), X2 = c(2, 4, 6, 8))
model_multi <- lm(y ~ X1 + X2, data = X_multi)

# Snippet 3: Assumptions and Diagnostics
residuals <- resid(model)
plot(predict(model), residuals)
abline(h = 0, col = "red")

# Snippet 4: Interpretation and Applications (Using the previous models)
slope <- coef(model)['X']
intercept <- coef(model)['(Intercept)']
